java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -981,179 +986,64 @@ public void shouldReturnAllPartitions() throws IOException, StateStoreException
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region2)
-                .id("id2")
-                .leafPartition(true)
-                .parentPartitionId(rootPartition.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        rootPartition = rootPartition.toBuilder()
-                .leafPartition(false)
-                .childPartitionIds(Arrays.asList(partition1.getId(), partition2.getId()))
-                .build();
-        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(rootPartition, partition1, partition2);
-        Region region3 = new Region(new RangeFactory(schema).createRange(field, 1L, 9L));
-        Partition partition3 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region3)
-                .id("id3")
-                .leafPartition(true)
-                .parentPartitionId(partition2.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        Region region4 = new Region(new RangeFactory(schema).createRange(field, 9L, null));
-        Partition partition4 = Partition.builder()
-                .rowKeyTypes(schema.getRowKeyTypes())
-                .region(region4)
-                .id("id4")
-                .leafPartition(true)
-                .parentPartitionId(partition2.getId())
-                .childPartitionIds(new ArrayList<>())
-                .dimension(-1)
-                .build();
-        partition2 = partition2.toBuilder()
-                .leafPartition(false)
-                .childPartitionIds(Arrays.asList(partition3.getId(), partition4.getId()))
-                .build();
-        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(partition2, partition3, partition4);
 
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .buildTree();
+        StateStore dynamoDBStateStore = inMemoryStateStoreWithPartitions(List.of(tree.getRootPartition()));
+        PartitionTree expectedTree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "id1", "id2", 1L)
+                .buildTree();
         // When
-        List<Partition> retrievedPartitions = dynamoDBStateStore.getLeafPartitions();
-        retrievedPartitions.sort((p1, p2) -> {
-            long p1Key = (long) p1.getRegion().getRange("key").getMin();
-            long p2Key = (long) p2.getRegion().getRange("key").getMin();
-            if (p1Key < p2Key) {
-                return -1;
-            } else if (p1Key == p2Key) {
-                return 0;
-            }
-            return 1;
-        });
+        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(expectedTree.getRootPartition(), expectedTree.getPartition("id1"), expectedTree.getPartition("id2"));
```

Comment:
This doesn't quite match the behaviour before. The root partition was split to partitions 1 and 2, then partition 2 was split to partitions 3 and 4.

Note that you'll need three partition trees to get the state of partition 2 right in the intermediate update.